Add MongoDB Atlas Metrics and Logs Documentation

Calm-Rock opened the pull request:
This PR adds comprehensive documentation for **MongoDB Atlas metrics and logs collection** with SigNoz, following the updated issue requirements that now include both metrics and logs.

## What's included:

### Documentation Structure
- **Overview**: Clarified approach - explains how to collect MongoDB Atlas telemetry using OpenTelemetry receiver
- **Prerequisites**: Focused on MongoDB Atlas requirements and API permissions
- **Setup**: Complete step-by-step guide starting with OpenTelemetry Collector installation
- **Dashboard**: Instructions for importing the MongoDB Atlas dashboard
- **Logs Explorer**: Documentation for logs collection and visualization
- **Metrics Reference**: Complete list of available MongoDB Atlas metrics
- **Troubleshooting**: Common issues and solutions for logs collection

### Key Features
- ✅ **Metrics Collection**: 25+ MongoDB Atlas metrics including database, performance, operations, and resource utilization
- ✅ **Logs Collection**: Comprehensive logs including:
  - **Audit logs** (database audit trail)
  - **Host logs** (mongod, mongos server logs)  
  - **Access logs** (authentication/authorization events)
  - **Project events** (configuration changes)
  - **Organization events** (admin actions)
- ✅ Uses official OpenTelemetry Collector (not distros)
- ✅ Supports both SigNoz Cloud and Self-hosted deployments
- ✅ Complete YAML configuration examples for metrics and logs
- ✅ Environment variables setup guide
- ✅ Storage extension configuration to prevent duplicate data
- ✅ Project and cluster filtering options
- ✅ Image placeholders for both metrics and logs screenshots

### Technical Details
- **File**: `data/docs/integrations/mongodb-atlas.mdx`
- **ID**: `mongodb-atlas`
- **Title**: MongoDB Atlas Metrics and Logs
- **Sidebar**: Will appear under integrations section

### Setup Structure Changes
- **Step 1**: Install OpenTelemetry Collector (moved from prerequisites)
- **Step 2**: Create collector config file (includes both metrics and logs)
- **Step 3**: Set environment variables (expanded for logs)
- **Step 4**: Run collector with config

### Configuration Highlights
- **Metrics**: Database size, operations counters, CPU/memory usage, connections, network I/O
- **Logs**: Audit logs, host logs, access logs with filtering and pagination options
- **Events**: Project and organization events with configurable polling
- **Storage**: File storage extension to prevent data duplication

### Dashboard Reference
References the dashboard location: `https://github.com/SigNoz/dashboards/blob/main/mongodb/mongodb-atlas-otlp-v1.json`

### Images Required
The documentation includes placeholders for the following screenshots:

**Metrics Dashboards:**
- `/img/docs/integrations/mongodb-atlas/cluster-overview.webp`
- `/img/docs/integrations/mongodb-atlas/database-metrics.webp`
- `/img/docs/integrations/mongodb-atlas/performance-metrics.webp`
- `/img/docs/integrations/mongodb-atlas/operations-metrics.webp`
- `/img/docs/integrations/mongodb-atlas/resource-utilization.webp`

**Logs Explorer:**
- `/img/docs/integrations/mongodb-atlas/audit-logs.webp`
- `/img/docs/integrations/mongodb-atlas/host-logs.webp`
- `/img/docs/integrations/mongodb-atlas/access-logs.webp`
- `/img/docs/integrations/mongodb-atlas/events-logs.webp`

### Prerequisites Documentation
- MongoDB Atlas account with API access
- API key permissions (Project Read Only, Organization Member roles)
- Audit logging enabled in Atlas (for audit logs)

### Troubleshooting Section
- Common logs collection issues and solutions
- Error message explanations
- Configuration verification steps

Resolves #1615

prashant-shahi commented on docs/userguide/sso-authentication.md:
```diff
@@ -82,3 +82,45 @@ For more details on the metadata page, click [here](https://learn.microsoft.com/
+#### Steps to be performed in Okta
+
+1. Log in to **Okta**
+2. From the **Admin** page, go to **Applications** > **Applications** > **Create App Integration**
```
It is correct

prashant-shahi commented on docs/userguide/sso-authentication.md:
```diff
@@ -82,3 +82,45 @@ For more details on the metadata page, click [here](https://learn.microsoft.com/
+    - **Single Sign-on URL**: `http(s)://${SIGNOZ_BASEURL}/api/v1/complete/saml`
+    - **Audience URI (SP Entity ID)**: `http(s)://${SIGNOZ_BASEURL}`
+5. Save the application integration
+6. Now, we can visit the **Metadata URL** in Okta to get the following information:
```
Added the gif

prashant-shahi commented on docs/userguide/sso-authentication.md:
```diff
@@ -82,3 +82,45 @@ For more details on the metadata page, click [here](https://learn.microsoft.com/
+3. Select **SAML 2.0**, and hit **Next**
+4. On the **SAML Integration** page, enter the following:
+    - **Application Name**: `SigNoz`
+    - **Single Sign-on URL**: `http(s)://${SIGNOZ_BASEURL}/api/v1/complete/saml`
```
`SIGNOZ_BASEURL` is consistent with the existing content in the same docs page.

This can be addressed in another PR where this is elaborated in all places where used.
Can you please pick this up?

Calm-Rock replied:
Okay @prashant-shahi. Will pick this up. 
Can you please create an issue for this and assign it to me.

prashant-shahi replied:
It looks like only this page has the exact same `SIGNOZ_BASEURL` used.
However, I remember somewhere we had base URL set up. It must have been main repo markdown docs, which we need not worry about too much for now.

Nvm, here is the issue link: https://github.com/SigNoz/signoz.io/issues/1679

prashant-shahi commented on docs/userguide/sso-authentication.md:
```diff
@@ -82,3 +82,45 @@ For more details on the metadata page, click [here](https://learn.microsoft.com/
+1. Log in to **Okta**
+2. From the **Admin** page, go to **Applications** > **Applications** > **Create App Integration**
+3. Select **SAML 2.0**, and hit **Next**
+4. On the **SAML Integration** page, enter the following:
```
Gif added

prashant-shahi commented on docs/userguide/sso-authentication.md:
````diff
@@ -82,3 +82,45 @@ For more details on the metadata page, click [here](https://learn.microsoft.com/
+    <md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="https://redacted.okta.com/app/redacted_signoz_1/exk...697/sso/saml"/>
+    ```
+
+#### Steps to be performed in SigNoz
````
GIF added

Calm-Rock committed:
Add MongoDB Atlas integration documentation

Calm-Rock committed:
Update MongoDB Atlas documentation to include logs collection

Calm-Rock commented:
## ✅ Documentation Updated Successfully!

The MongoDB Atlas documentation has been updated to address all the feedback and align with the updated issue requirements:

### 🔄 **Key Changes Made:**

#### **📊 Expanded Coverage**
- **Title**: Updated to "**MongoDB Atlas Metrics and Logs**" 
- **Scope**: Now covers comprehensive **metrics AND logs collection**

#### **📝 Improved Overview**
- ✅ **Fixed**: Removed misleading "out-of-the-box dashboard" language
- ✅ **Clarified**: This explains how to collect MongoDB Atlas telemetry using OpenTelemetry receiver
- ✅ **Accurate**: Mentions there is a dashboard available for use (not claiming it comes automatically)

#### **⚙️ Restructured Setup Process**
- ✅ **Step 1**: **OpenTelemetry Collector installation** (moved from prerequisites as requested)
- ✅ **Step 2**: Create collector config file (now includes both metrics and logs)
- ✅ **Step 3**: Set environment variables (expanded for logs)
- ✅ **Step 4**: Run collector with config

#### **📋 Enhanced Prerequisites**
- ✅ **Focused**: Only MongoDB Atlas account and API requirements
- ✅ **Specific**: API key permissions (Project Read Only, Organization Member)
- ✅ **Clear**: Audit logging requirement for audit logs

#### **🗂️ Complete Logs Documentation**
Added comprehensive logs collection including:
- **🔍 Audit Logs**: Database audit trail (requires Atlas audit logging)
- **📋 Host Logs**: mongod and mongos server logs
- **🔐 Access Logs**: Authentication/authorization events with filtering
- **📅 Project Events**: Configuration changes and project activities
- **🏢 Organization Events**: Admin actions and org-level changes

#### **📊 Technical Enhancements**
- ✅ **Storage Extension**: File storage to prevent duplicate data
- ✅ **Project Filtering**: Include/exclude clusters configuration
- ✅ **Pagination**: Configurable page sizes and limits for API calls
- ✅ **Polling Intervals**: Customizable collection frequencies
- ✅ **Error Handling**: Comprehensive troubleshooting section

#### **🖼️ Image Placeholders**
Added placeholders for **9 screenshots**:
- **5 Metrics Dashboards**: Cluster overview, database metrics, performance, operations, resource utilization
- **4 Logs Views**: Audit logs, host logs, access logs, events in SigNoz Logs Explorer

#### **🔧 Complete Configuration**
- **Metrics**: 25+ MongoDB Atlas metrics with detailed descriptions
- **Logs**: Full YAML configuration with all log types
- **Storage**: File storage extension setup
- **Exporters**: Both SigNoz Cloud and self-hosted options

The documentation now provides **complete coverage** of MongoDB Atlas observability capabilities and follows the exact structure and requirements outlined in the updated issue!

Calm-Rock commented:
Docs here https://signoz.io/docs/integrations/mongodb-atlas/
